plugins/pulp_python/plugins/distributors/steps.py

```diff
@@ -60,67 +59,125 @@ def process_main(self):
-        package_path = os.path.join(simple_path, name)
-        os.makedirs(package_path)
-        package_index_path = os.path.join(package_path, 'index.html')
+        :param packages: A list of packages
+        :type  packages: list of pulp_python.plugins.models.Package
+        """
+        project_path = os.path.join(simple_path, project_name)
+        os.makedirs(project_path)
+        package_index_path = os.path.join(project_path, 'index.html')
         with open(package_index_path, 'w') as package_index:
             html = ElementTree.Element('html')
             head = ElementTree.SubElement(html, 'head')
             title = ElementTree.SubElement(head, 'title')
-            title.text = 'Links for %s' % name
+            title.text = 'Links for %s' % project_name
             ElementTree.SubElement(head, 'meta', {'name': 'api-version', 'value': '2'})
             body = ElementTree.SubElement(html, 'body')
             heading = ElementTree.SubElement(body, 'h1')
             heading.text = title.text
             for package in packages:
-                href = '../../%s#%s=%s' % (_get_package_path(name, package['filename']),
-                                           package['checksum_type'], package['checksum'])
+                href = '../../packages/%s' % package.checksum_url
                 node = ElementTree.SubElement(body, 'a', {'href': href, 'rel': 'internal'})
                 node.text = package['filename']
                 ElementTree.SubElement(body, 'br')
 
             package_index.write(ElementTree.tostring(html, 'utf8'))
 
+    def write_json_api(self, projects):
+        """
+        Create a json file for each project in the repository.
+
+        The structure of the data is designed to mimic the json api of PyPI. The data will be
+        for a single Python project (eg. SciPy). The inner dictionary 'info' specifies details of
+        the project that should be applicable for all packages. The inner dictionary 'releases'
+        contains keys for each version and the value is a list of dictionaries, each representing
+        the metadata for a single package of that version.
+
+        More information on the PyPI API can be found here: https://wiki.python.org/moin/PyPIJSON
+
+        :param projects: keys are project names, values are lists of packages of that project
+        :type  projects: dict
+        """
+        api_path = os.path.join(self.parent.web_working_dir, 'pypi')
+        for project_name, packages in projects.items():
+            project_metadata_path = os.path.join(api_path, project_name, 'json')
+            os.makedirs(project_metadata_path)
+            project_index_metadata_path = os.path.join(project_metadata_path, 'index.json')
+            with open(project_index_metadata_path, 'w') as project_json:
+                data = PublishMetadataStep._create_project_metadata(project_name, packages)
+                project_json.write(json.dumps(data))
```

Comment:
This works, but ``json`` provides the ``dump`` method which does the same thing. It's also nice if people will ever look at this metadata to format it prettily. This is purely my preference and you can take it or leave it :shipit:.

``json.dump(data, project_json, sort_keys=True, indent=4, separators=(',', ': '))``